### PR TITLE
Update to address omissions

### PR DESCRIPTION
**Pull request by BillWagner**
See issue #4103

This PR updates the topic on ref locals and returns to address a number of omissions in the earlier version.

[Internal review link](https://review.docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/ref-returns?branch=pr-en-us-4194)

### CODE REVIEW

**Comment by BillWagner** [comment]
I think this is ready.  @ericlippert do you want to review the changes?

File: docs/csharp/programming-guide/classes-and-structs/ref-returns.md [hunk 1]
```diff
@@ -3,40 +3,35 @@ title: "Ref return values and ref locals (C# Guide)"
...
+Starting with C# 7, C# supports reference return values (ref returns). A reference return value allows a method to return a reference to a variable, rather than a value, back to a caller. The caller can then choose to treat the returned variable as if it were returned by value or by reference. The caller can create a new variable that is itself a reference to the returned value, called a ref local.
 
 ## What is a reference return value?
 
-Most developers are familiar with passing an argument to a called method *by reference*. A called method's argument list includes a value passed by reference, and any changes made to its value by the called method are returned to the caller. A *reference return value* is the opposite:
+Most developers are familiar with passing an argument to a called method *by reference*. A called method's argument list includes a variable passed by reference, and any changes made to its value by the called method are observed by the caller. A *reference return value* means that a method returns a *reference* (or an alias) to some variable whose scope includes the method and whose lifetime must extend beyond the return of the method.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/classes-and-structs/ref-returns.md]: It's somewhat unclear what the following list is supposed to represent. There's no transition to it from this paragraph.

File: docs/csharp/programming-guide/classes-and-structs/ref-returns.md [hunk 2]
```diff
@@ -3,40 +3,35 @@ title: "Ref return values and ref locals (C# Guide)"
...
+- The return value cannot be the literal `null`. Attempting to return `null` generates compiler error CS8156, "An expression cannot be used in this context because it may not be returned by reference."
 
-   If a method with a ref return needs to return a null value, you can either return a null (uninstantiated) value for a reference type or a [nullable type](../nullable-types/index.md) for a value type.
+   A method with a ref return can return an alias to a variable whose value is currently the null (uninstantiated) value or a [nullable type](../nullable-types/index.md) for a value type.
  
-- The return value cannot be a constant, an enumeration member, or a property of a `class` or `struct`. Attempting to return these generates compiler error CS8156, "An expression cannot be used in this context because it may not be returned by reference."
+- The return value cannot be a constant, an enumeration member, or the by-value return value from a property, or a method of a `class` or `struct`. Attempting to return these generates compiler error CS8156, "An expression cannot be used in this context because it may not be returned by reference."
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/classes-and-structs/ref-returns.md]: or the by-value return value --> the by-value return value (there is one too many "or" here)

**Review by rpetrusha — APPROVED** [review]
Thanks for doing this, @BIllWagner; I realized somewhere in the middle of reading it that I originally authored this topic and so was responsible for the inaccuracies. In any case, your revision looks good; there are just two minor comments. Once you address them, feel free to merge.

**Pull request merged**